### PR TITLE
Fix overflow

### PR DESCRIPTION
**Pull request by LorenzoFritzsch**
Fixes overflow when initializing any `TimeAmount` other than `TimeAmount.nanoseconds` to `Int64.max`.

### Motivation:

Closes https://github.com/apple/swift-nio/issues/2532

### Modifications:

Added a check to the amount calculated in nanoseconds. If it overflows, `TimeAmount(.max)` is returned, otherwise `TimeAmount(nanoseconds)`.

### Result:
Program doesn't crash when setting a `TimeAmount` to be `Int64.max`. Instead, `TimeAmount(.max)` is returned.

### CODE REVIEW

File: Sources/NIOCore/EventLoop.swift [hunk 1]
```diff
@@ -436,49 +436,64 @@ public struct TimeAmount: Hashable, Sendable {
     /// - parameters:
     ///     - amount: the amount of microseconds this `TimeAmount` represents.
     /// - returns: the `TimeAmount` for the given amount.
+    ///
+    /// - note: returns `TimeAmount(.max)` if the amount, once converted to nanoseconds, exceeds `Int64.max`.
     @inlinable
     public static func microseconds(_ amount: Int64) -> TimeAmount {
-        return TimeAmount(amount * 1000)
+        let nanoseconds = amount &* 1000
+        return nanoseconds >= 0 ? TimeAmount(nanoseconds) : TimeAmount(.max)
```
Comment (glbrntt) [inline Sources/NIOCore/EventLoop.swift:444]: This isn't quite right: `TimeAmount` can represent a negative duration so `.microseconds(-1000)` would be incorrectly converted to `.max`. We'd also need to have unit tests for the extreme (overflow and underflow) cases to verify this behaves as expected.

**Review by glbrntt — CHANGES_REQUESTED** [review]

**Review by Lukasa — CHANGES_REQUESTED** [review]
As noted by @weissi 's comment in the linked issue, it would also be better to replace the unchecked operations by `reportingOverflow` methods.

File: Tests/NIOPosixTests/EventLoopTest.swift [hunk 2]
```diff
@@ -1608,6 +1608,24 @@ public final class EventLoopTest : XCTestCase {
...
+    }
+    
+    func testTimeAmountCappedUnderflow() {
+        let underflowCap = TimeAmount(.min)
+        XCTAssertEqual(TimeAmount.microseconds(.min), underflowCap)
+        XCTAssertEqual(TimeAmount.milliseconds(.min), underflowCap)
+        XCTAssertEqual(TimeAmount.seconds(.min), underflowCap)
+        XCTAssertEqual(TimeAmount.minutes(.min), underflowCap)
+        XCTAssertEqual(TimeAmount.hours(.min), underflowCap)
+    }
```
Comment (glbrntt) [inline Tests/NIOPosixTests/EventLoopTest.swift:1628]: Sorry to be a pain, but could you move these to https://github.com/apple/swift-nio/blob/main/Tests/NIOCoreTests/TimeAmountTests.swift?

**Comment by glbrntt** [comment]
@swift-server-bot test this please

**Review by glbrntt — APPROVED** [review]

**Review by Lukasa — APPROVED** [review]
Nice, this looks really good to me. Let's get the tests running.

**Comment by Lukasa** [comment]
@swift-server-bot test this please
